test/cli/test-other.py

```diff
@@ -825,3 +826,41 @@ def test_file_duplicate_2(tmpdir):
         '3/3 files checked 0% done'
     ]
     assert stderr == ''
+
+
+def test_premium_with_relative_path(tmpdir): #12254
+
+
+    test_file = os.path.join(tmpdir, 'test.c')
+    with open(test_file, 'wt'):
+        pass
+
+    product_name = 'Cppcheck Premium ' + str(time.time())
+
+    test_cfg = lookup_cppcheck_exe()
+    if(test_cfg.endswith('.exe')):
+        test_cfg = test_cfg[:-4]
+    test_cfg += '.cfg'
+    with open(test_cfg, 'wt') as f:
+        f.write("""
+                {
+                    "addons": [],
+                    "productName": "NAME",
+                    "about": "Cppcheck Premium 1.2.3.4"
+                }
+                """.replace('NAME', product_name))
+
+    if os.path.isfile('cppcheck'):
+        os.chdir('test/cli')
```

Comment:
Just no. If you need to change the directory for running a process please pass the `cwd` parameter to the call. `cppcheck()` might already expose that. If it doesn't please pass that through as a an additional positional parameter.